gax/src/main/java/com/google/api/gax/longrunning/OperationFutureImpl.java

```diff
@@ -81,6 +81,22 @@ public OperationFutureImpl(
     this.metadataTransformer = checkNotNull(metadataTransformer);
   }
 
+  public OperationFutureImpl(
+      RetryingFuture<OperationSnapshot> pollingFuture,
+      ApiFuture<OperationSnapshot> initialFuture,
+      ApiFunction<OperationSnapshot, ResponseT> responseTransformer,
+      ApiFunction<OperationSnapshot, MetadataT> metadataTransformer,
+      ApiFunction<Exception, ResponseT> exceptionTransformer) {
+    this.pollingFuture = checkNotNull(pollingFuture);
```

Comment:
I think we need to `ApiFutures.catching` here as well. If we look at `get()` implementation, it also waits for `pollingFuture` to complete. If `pollingFuture` fails, then waiting for it would throw exception.

Reply:
Synced offline: pollingFuture is not a `RetryingFuture` and thus cannot be returned by `ApiFutures.catching`